### ISSUE TITLE
Journal commands to a .sty file?

### ISSUE THREAD

**Issue by yoachim** [AASJournals/AASTeX60#86]
I seem to often be in the situation where I want to use the usual journal commands for bibtex to work, but don't necessarily want my document to use the full aastex document class. Is there any reason not to break the journal and other common commands (e.g. `\sun`) out into their own .sty file so they can be used with any document class?

**Comment by gregschwarz** [comment]
I am not sure I understand. Are you asking if you can create your own .sty file with the standard astronomy commands like \sun and use it with any non-AASTeX classfile? If so then the answer is yes. There is no requirement to use AASTeX and it is open source so you can take what you need.

**Comment by yoachim** [comment]
Right, I know everyone just copy-pastas those commands all over the place. But it might be nice if this repo had a .sty file with the commands so they were maintained and easy to use. And presumably then you wouldn't want the code in both the .cls and the .sty, so it would be a bit of refactoring.

**Comment by pkgw** [comment]
I'll just chime in to say I also think this would be nice to have. But I don't have the bandwidth to volunteer to put together a PR ...

**Comment by gregschwarz** [comment]
I still need more input on this. AASTeX is full of many "commands". Are you talking about big things like deluxetable or just your run of the mill definitions like these:

\newcommand\arcdeg{\mbox{$^\circ$}}% 
\newcommand\arcmin{\mbox{$^\prime$}}% 
\newcommand\arcsec{\mbox{$^{\prime\prime}$}}% 
\newcommand\fd{\mbox{$.\!\!^{\mathrm d}$}}% 
\newcommand\fh{\mbox{$.\!\!^{\mathrm h}$}}% 
\newcommand\fm{\mbox{$.\!\!^{\mathrm m}$}}% 
\newcommand\fs{\mbox{$.\!\!^{\mathrm s}$}}% 
\newcommand\fdg{\mbox{$.\!\!^\circ$}}% 

If it is the latter then it should be simple to egrep out all the \newcommands, remove the ones you do not want, and dump the rest into a new .sty file.

**Comment by augustfly** [comment]
Sidebar: I would advocate for *not* producing a stand alone `deluxetable`. the bugs in that thing are so probable and the dependencies non-trivial that we'd never keep caught up on them if we start officially forking support. 

@gregschwarz They are specifically talking about `\aj, \apj,` etc. I do think there is some merit in acknowledging that the journal macros get wide usage outside of AASTeX. AFAIK we don't officially maintain them, and there are innumerable RYO latex packages that exist only add on missing ones. For instance "likeapj" states:

> "Astronomy Letters (AstL) and Astronomische Nachrichten (AN) need \anl and \anach in the journal entry of the bib file."


The first google result of someone rolling their own such file is from CDS: http://cdsads.u-strasbg.fr/abs_doc/aas_macros.html and http://cdsads.u-strasbg.fr/abs_doc/aas_macros.sty
So I would support pulling the journal macros into their own file, which is subsequently called by AASTeX instead of having them embedded. A problem with this precedent is that AASTeX is such a mess of commands internally that there might be strong dependencies or a temptation to start de-crufting it of newcommands in other ways. 

A hard part is defining what is the AAS's purview versus "things widely supported by TeX" or another third party package.  For instance in this documentation there are symbols referenced that I think we can control and many we do not!

https://journals.aas.org/aastex-symbol/

which depends upon

`\RequirePackage{amssymb}% `

etc.

**Comment by joeharr4** [comment]
This one has been dormant for a while, but I'd like to bring it back up.

I think you can separate aastex into "things that make it look like the journal" and "things that are useful in other formats".  Journal macros, symbols, and yes, the deluxetable would all be very useful in their own file.  They are separate from the numbering and style of sections, the formatting of a page, fonts, and so on.  Perhaps you are worried about deluxetable and want to hold that out for now, but I'd advocate for separating as much as possible of the things useful in other formats into their own file, making that the only place they exist, and importing that in aastex.  Call it astrostuff.sty or something.  You'd get the thanks and adulation of astronomers worldwide!

--jh--
